swarm.go

```diff
@@ -374,11 +374,24 @@ func (s *Swarm) Start(srv *p2p.Server) error {
 	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
 
 	if s.config.SwapEnabled {
-		err := s.swap.Deploy(context.Background(), s.backend, s.config.Path)
-		if err != nil {
-			return fmt.Errorf("Unable to deploy swap contract: %v", err)
+		if s.config.Contract != (common.Address{}) {
+			address := s.config.Contract
+			err := cswap.ValidateCode(context.Background(), s.backend, address)
```

Comment:
I would recommend using this if style to be consistent with the rest of the codebase (same for the other 3 error checks):
```go
if err := cswap.ValidateCode(context.Background(), s.backend, address); err != nil {
	return fmt.Errorf("Provided address not a chequebook smart-contract: %v", err)
}
```

Reply:
Done!